bibtexparser/bibdatabase.py

```diff
@@ -103,6 +116,48 @@ def expand_string(self, name):
         except KeyError:
             raise(UndefinedString(name))
 
+    def _add_missing_from_crossref_entry(self, entry, dependencies=set()):
+        if entry['ID'] in self._crossref_updated:
+            return
+
+        if entry['_crossref'] not in self._entries_dict:
+            logger.error("Crossref reference %s for %s is missing.",
+                         entry['_crossref'],
+                         entry['ID'])
+            return
+
+        if entry['_crossref'] in dependencies:
+            logger.error("Circular crossref dependency: %s->%s->%s.",
+                         "->".join(dependencies),
+                         entry['ID'],
+                         entry['_crossref'])
+            return
```

Comment:
Same thing: we might want to just fail.

Reply:
I think logging would be more reasonable then failing because the entry is valid, and it only does not resolve an unnecessary crossref properly. If we visit the same entry through a crossref chain again, then we already updated every field of the current entry with it (at the earlier visit), so we can't get anything "new." In turn, we can just stop here, which is what we do.

Reply:
I marking the conversation as resolved, so that GitHub updates the PR correctly.